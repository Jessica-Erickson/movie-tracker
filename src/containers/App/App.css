@import url('https://fonts.googleapis.com/css?family=Roboto+Slab');

header {
  margin: 0;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  align-items: center;
  grid-template-areas: 'b a e' 'c . .' 'd . .';
  height: 100px;
  background: #990000;
  padding: 10px;
}
.title {
  grid-area: a;
  margin: 0;
  justify-self: center;
  text-decoration: none;
  font-family: 'Roboto Slab', serif;
  color: #CACAD1;
  font-size: 3em;
  text-shadow: -5px 0 4px black, 0 1px black, 1px 0 black, 0 -1px black;
}
h2 {
  margin: 0;
  grid-area: b;
}
.favorites {
  grid-area: e;
}
.log-link {
  grid-area: c;
  text-decoration: none;
  color: #CACAD1;
  text-shadow: 3px 2px black, 1px 2px 2px black;
  font-size: 1.5em;
}
.fa-sign-in-alt {
  margin-left: 1%;
}
form {
  grid-area: d;
}