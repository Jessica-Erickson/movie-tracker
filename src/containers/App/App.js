import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';
import CardContainer from '../CardContainer/CardContainer'
import './App.css';
import SignUp from '../Authentication/SignUp/SignUp';
import SignIn from '../Authentication/SignIn/SignIn'

class App extends Component {

  render() {
    const { id , name } = this.props;
    return (
      <div className="App">
        {id ? <h2>{`Welcome back, ${name}!`}</h2> : <h2>Please sign in</h2>}
        <SignIn />
        <SignUp />
        <Route exact path='/' component={ CardContainer }/>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  id: state.id,
  name: state.name,
})

export default connect(mapStateToProps , null)(App);
