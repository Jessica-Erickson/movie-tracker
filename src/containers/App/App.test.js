import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {Provider} from 'react-redux'
import {BrowserRouter} from 'react-router-dom';
import { createStore } from 'redux'
import configureMockStore from 'redux-mock-store'

it('renders without crashing', () => {
  const mockStore = configureMockStore()
  const initialState = {movies: []}
  const store = mockStore(initialState);
  const div = document.createElement('div');
  ReactDOM.render(<Provider store={store}>
    <BrowserRouter>
      <App />
    </BrowserRouter>
  </Provider>, div);
  ReactDOM.unmountComponentAtNode(div);
});
