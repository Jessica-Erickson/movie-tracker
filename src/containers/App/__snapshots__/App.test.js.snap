// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <Connect(SignIn) />
  <Connect(SignUp) />
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
</div>
`;
